bigtable-hbase-parent/bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java

```diff
@@ -819,93 +836,172 @@ public boolean isTableAvailable(TableName tableName, byte[][] splitKeys) throws
   @Override
   public void snapshot(String snapshotName, TableName tableName)
       throws IOException, SnapshotCreationException, IllegalArgumentException {
-    throw new UnsupportedOperationException("snapshot");  // TODO
+    com.google.longrunning.Operation operation = snapshotTable(snapshotName, tableName);
```

Comment:
currentOperationState.getDone() isn't enough.  You still need to check for the resultCase.  Your point about checkout the operation that's returned is well taken.  No need to look it up again in the quick cases.